src/pybind/mgr/orchestrator/module.py

```diff
@@ -1140,27 +1112,21 @@ def _apply_misc(self,
             self._orchestrator_wait([completion])
             raise_if_exception(completion)
             data = completion.result
-            if format == 'plain':
+            if format == Format.plain:
                 out = generate_preview_tables(data)
             else:
                 out = to_format(data, format, many=True, cls=None)
         return HandleCommandResult(stdout=out)
 
-    @_cli_write_command(
-        'orch apply mds',
-        'name=fs_name,type=CephString '
-        'name=placement,type=CephString,req=false '
-        'name=dry_run,type=CephBool,req=false '
-        'name=unmanaged,type=CephBool,req=false '
-        'name=format,type=CephChoices,strings=plain|json|json-pretty|yaml,req=false',
-        'Update the number of MDS instances for the given fs_name')
+    @_cli_write_command('orch apply mds')
     def _apply_mds(self,
                    fs_name: str,
                    placement: Optional[str] = None,
                    dry_run: bool = False,
                    unmanaged: bool = False,
```

Comment:
```suggestion
                   dry_run: Optional[bool] = False,
                   unmanaged: Optional[bool] = False,
```

Btw (independent to this usage here), using `Optional[bool]` is typical an antipattern when using types, except if you **really** want to have `None` values in addition to `False` and `True`.  This is the reason I typically used standard non-optional bool annotations for optional bool parameters

Reply:
so the recommended way is plain `dry_run: bool = False`, right?  i can come up with a change which crosses check the `default` field and `annotations` field in a follow-up PR.

```
In [1]: def foo(dry: bool = False):
   ...:     pass
   ...:

In [2]: import inspect

In [3]: inspect.getfullargspec(foo)
Out[3]: FullArgSpec(args=['dry'], varargs=None, varkw=None, defaults=(False,), kwonlyargs=[], kwonlydefaults=None, annotations={'dry': <class 'bool'>})
```

Reply:
In my opinion `Optional[bool]` in this specific use case is perfectly fine. It's just that you should avoid it in general.

Reply:
@sebastian-philipp fixed anyway.